packages/frontend/src/hooks/useCampaign.ts

```diff
@@ -0,0 +1,58 @@
+import type { SupportedChain } from "@metrom-xyz/contracts";
+import { type Campaign } from "@metrom-xyz/sdk";
+import { useEffect, useState } from "react";
+import type { Address } from "viem";
+import { metromApiClient, SUPPORTED_AMM_SLUG_TO_NAME } from "../commons";
+
+export interface NamedCampaign extends Campaign {
+    name: string;
+}
+
+export function useCampaign(
+    chainId: SupportedChain,
+    id: Address,
+): {
+    loading: boolean;
+    campaign?: NamedCampaign;
+} {
+    const [campaign, setCampaign] = useState<NamedCampaign>();
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        let cancelled = false;
+
+        async function fetchData() {
+            if (!cancelled) setLoading(false);
+            if (!cancelled) setCampaign(undefined);
+
+            try {
+                if (!cancelled) setLoading(true);
+                const campaign = await metromApiClient.fetchCampaign({
+                    chainId,
+                    id,
+                });
+
+                if (!cancelled)
+                    setCampaign({
+                        ...campaign,
+                        name: `${SUPPORTED_AMM_SLUG_TO_NAME[campaign.pool.amm] || "-"} ${campaign.pool.token0.symbol} / ${campaign.pool.token1.symbol}`,
```

Comment:
We could extract this to its own utils function

Reply:
Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.